samples/subsys/fs/nffs/src/main.c

```diff
@@ -0,0 +1,149 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <fs.h>
+#include <nrf.h>
+#include <device.h>
+#include <gpio.h>
+#include <board.h>
+
+int lock=0;
```

Comment:
What is going on with this variable? I think you might want to be using a struct k_sem instead.